app/src/main/java/com/anytypeio/anytype/ui/date/DateObjectFragment.kt

```diff
@@ -0,0 +1,265 @@
+        ) {
+            composable(route = DATE_MAIN) {
+                DateMainScreen(
+                    uiCalendarIconState = vm.uiCalendarIconState.collectAsStateWithLifecycle().value,
+                    uiSyncStatusBadgeState = vm.uiSyncStatusBadgeState.collectAsStateWithLifecycle().value,
+                    uiHeaderState = vm.uiHeaderState.collectAsStateWithLifecycle().value,
+                    uiFieldsState = vm.uiFieldsState.collectAsStateWithLifecycle().value,
+                    uiObjectsListState = vm.uiObjectsListState.collectAsStateWithLifecycle().value,
+                    uiNavigationWidget = vm.uiNavigationWidget.collectAsStateWithLifecycle().value,
+                    uiFieldsSheetState = vm.uiFieldsSheetState.collectAsStateWithLifecycle().value,
+                    uiContentState = vm.uiContentState.collectAsStateWithLifecycle().value,
+                    canPaginate = vm.canPaginate.collectAsStateWithLifecycle().value,
+                    uiCalendarState = vm.uiCalendarState.collectAsStateWithLifecycle().value,
+                    uiSyncStatusState = vm.uiSyncStatusWidgetState.collectAsStateWithLifecycle().value,
+                    onDateEvent = vm::onDateEvent
+                )
+            }
+        }
+    }
+
+    @OptIn(ExperimentalMaterial3Api::class)
+    @Composable
+    private fun ErrorScreen() {
+        val errorStateScreen = vm.errorState.collectAsStateWithLifecycle()
+        when (val state = errorStateScreen.value) {
+            UiErrorState.Hidden -> {
+                // Do nothing
+            }
+            is UiErrorState.Show -> {
+                val message = when (val r = state.reason) {
+                    is UiErrorState.Reason.YearOutOfRange ->
+                        stringResource(
+                            id = R.string.date_layout_alert_date_out_of_range,
+                            r.min,
+                            r.max
+                        )
+                    is UiErrorState.Reason.ErrorGettingFields -> r.msg
+                    is UiErrorState.Reason.ErrorGettingObjects -> r.msg
+                    is UiErrorState.Reason.Other -> r.msg
+                }
+                BaseAlertDialog(
+                    dialogText = message,
+                    buttonText = stringResource(id = R.string.membership_error_button_text_dismiss),
+                    onButtonClick = vm::hideError,
+                    onDismissRequest = vm::hideError
+                )
+            }
+        }
+    }
+
+    override fun injectDependencies() {
+        val params = DateVmParams(
+            spaceId = SpaceId(space),
+            objectId = objectId
+        )
+        componentManager().dateObjectComponent.get(params).inject(this)
+    }
+
+    override fun releaseDependencies() {
+        componentManager().dateObjectComponent.release()
+    }
+
+    override fun onApplyWindowRootInsets(view: View) {
+        // Do nothing. TODO add ime padding.
```

Comment:
may need to check on older android versions

Reply:
will do